packages/account-v2/src/components/LeaveConfirm/__tests__/LeaveConfirm.spec.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { Formik } from 'formik';
+import { Modal } from '@deriv-com/ui';
+import { act, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { LeaveConfirm } from '../LeaveConfirm';
+
+jest.mock('@deriv-com/ui', () => ({
+    ...jest.requireActual('@deriv-com/ui'),
+    useDevice: jest.fn(() => ({ isMobile: false })),
+}));
+
+const mockUnblock = jest.fn();
+let mockBlockFn = jest.fn();
+
+jest.mock('react-router-dom', () => ({
+    useHistory: jest.fn(() => ({
+        block: jest.fn(fn => {
+            mockBlockFn = fn;
+            return mockUnblock;
+        }),
+        push: jest.fn(),
+    })),
+}));
+
+const wrapper = ({ children }: { children: React.JSX.Element }) => {
+    return (
+        <Formik initialValues={{ name: '' }} onSubmit={jest.fn()}>
+            {({ handleBlur, handleChange, handleSubmit, values }) => (
+                <form onSubmit={handleSubmit}>
+                    {children}
+                    <input
+                        aria-label='name'
+                        name='name'
+                        onBlur={handleBlur}
+                        onChange={handleChange}
+                        value={values.name}
+                    />
+                </form>
+            )}
+        </Formik>
+    );
+};
+
+afterEach(() => {
+    jest.clearAllMocks();
+});
+
+describe('LeaveConfirm', () => {
+    const onCancelMock = jest.fn();
+    const onLeaveMock = jest.fn();
+
+    const renderComponent = () => {
+        render(<LeaveConfirm onCancel={onCancelMock} onLeave={onLeaveMock} />, { wrapper });
+        const inputField = screen.getByRole('textbox');
+        userEvent.type(inputField, 'Hello');
+
+        act(() => {
```

Comment:
may I clarify, why do we need here this act() ?

Reply:
because mockBlockfn is updating component state, and react-testing-library was throwing a warning here to use act.

Reply:
alright, but what is the sense of testing, if everything is mocked and triggered by us in test suite?
may I suggest you to not mock useHistory and not invoke modal deliberately,  but simulate real behaviour of user.

Reply:
```
it('should render the component 1', async () => {
        const history = createBrowserHistory();
        render(
            <Router history={history}>
                <Formik initialValues={{ name: '' }} onSubmit={jest.fn()}>
                    {({ handleBlur, handleChange, handleSubmit, values }) => (
                        <form onSubmit={handleSubmit}>
                            <LeaveConfirm onCancel={onCancelMock} onLeave={onLeaveMock} />
                            <input
                                aria-label='name'
                                name='name'
                                onBlur={handleBlur}
                                onChange={handleChange}
                                value={values.name}
                            />
                        </form>
                    )}
                </Formik>
            </Router>)

        const inputField = screen.getByRole('textbox');
        userEvent.type(inputField, 'Hello');

        await waitFor(() => {
            history.push('/something');
        })
        expect(screen.getByText(unsavedChangesText)).toBeInTheDocument();
        expect(
            screen.getByText(
                'You have unsaved changes. Are you sure you want to discard changes and leave this page?'
            )
        ).toBeInTheDocument();
    });
```

Reply:
May I ask you to try this approach for other suites as well 🙏 
you can also refactor and move rendering to separate function and reuse 🙏

Reply:
Updated.